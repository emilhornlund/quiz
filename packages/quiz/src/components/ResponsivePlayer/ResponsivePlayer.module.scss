@use '../../styles/colors';
@use '../../styles/helpers';
@use '../../styles/variables';

.playerWrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  overflow: hidden;
  border-style: solid;
  border-color: colors.$white-1-50;
  background-color: colors.$white-1-50;

  @include helpers.mobile {
    border-width: helpers.pxToRem(2px);
    border-radius: variables.$mobile-narrow-layout-gutter;
  }

  @include helpers.tablet {
    border-width: helpers.pxToRem(3px);
    border-radius: variables.$tablet-narrow-layout-gutter;
  }

  @include helpers.desktop {
    border-width: helpers.pxToRem(4px);
    border-radius: variables.$desktop-narrow-layout-gutter;
  }
}
