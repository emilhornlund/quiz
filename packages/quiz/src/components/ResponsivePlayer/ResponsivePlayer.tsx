import React, { FC } from 'react'
import ReactPlayer from 'react-player'

import styles from './ResponsivePlayer.module.scss'

export interface ResponsivePlayerProps {
  url: string
  playing?: boolean
  grow?: boolean
}

const ResponsivePlayer: FC<ResponsivePlayerProps> = ({
  url,
  playing = true,
  grow = true,
}) => (
  <div className={styles.playerWrapper}>
    <ReactPlayer
      src={url}
      muted={false}
      width={grow ? '100%' : undefined}
      height={grow ? '100%' : undefined}
      playing={playing}
      loop
      controls
    />
  </div>
)

export default ResponsivePlayer
