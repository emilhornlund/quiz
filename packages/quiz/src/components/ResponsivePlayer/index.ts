export type { ResponsivePlayerProps } from './ResponsivePlayer'
export { default } from './ResponsivePlayer'
