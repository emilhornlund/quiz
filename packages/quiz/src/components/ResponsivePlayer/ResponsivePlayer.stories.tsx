import type { Meta, StoryObj } from '@storybook/react'

import ResponsivePlayer from './ResponsivePlayer'

const meta = {
  component: ResponsivePlayer,
} satisfies Meta<typeof ResponsivePlayer>

export default meta
type Story = StoryObj<typeof meta>

export const Video = {
  args: {
    url: 'https://www.youtube.com/watch?v=LXb3EKWsInQ',
  },
} satisfies Story
