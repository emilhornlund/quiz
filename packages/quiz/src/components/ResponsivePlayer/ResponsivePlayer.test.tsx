import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import ResponsivePlayer from './ResponsivePlayer'

describe('ResponsiveImage', () => {
  it('should render ResponsiveImage with props', async () => {
    const { container } = render(
      <ResponsivePlayer url="https://www.youtube.com/watch?v=LXb3EKWsInQ" />,
    )
    expect(container).toMatchSnapshot()
  })
})
