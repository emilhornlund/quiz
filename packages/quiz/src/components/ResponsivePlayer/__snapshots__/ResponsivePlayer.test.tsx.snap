// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResponsiveImage > should render ResponsiveImage with props 1`] = `
<div>
  <div
    class="playerWrapper"
  />
</div>
`;
